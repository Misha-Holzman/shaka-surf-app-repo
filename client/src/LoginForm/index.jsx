import React     from 'react';
import PropTypes from 'prop-types';
import {
  Field, Control, Input, Button, Hero,
  HeroBody,
  Container,
  Columns,
  Column,
  Box,
  Title,
  Subtitle,
} from 'bloomer';


export default class TaskForm extends React.Component {
  constructor(props) {
    super(props);

    this.state = {
      email:    '',
      password: '',
    };

    this.handleChange = this.handleChange.bind(this);
    this.handleSubmit = this.handleSubmit.bind(this);
  }

  // as found in the docs
  // @see https://reactjs.org/docs/forms.html#handling-multiple-inputs
  handleChange(evt) {
    // destructure the event
    const { name, value } = evt.target;

    // set the state as needed
    this.setState({
      [name]: value,
    });
  }

  async handleSubmit(evt) {
    evt.preventDefault();
    const { submitAction } = this.props;
    const { email, password } = this.state;
    await submitAction({ email, password });

    // reset the form
    this.setState({
      email:    '',
      password: '',
    });
  }

  render() {
    const { buttonText, size, isGrouped } = this.props;
    const { email, password } = this.state;
    return (
      <Hero isColor="light" isFullHeight style={{ 'margin-top': '-60px'}}>
        <HeroBody>
          <Container hasTextAlign="centered">
            <Columns>
              <Column isSize={4} isOffset={4}>
                <Box>
                  <Title>Hiya!</Title>
                  <Subtitle>Letâ€™s Get You logged in...</Subtitle>
                  <form onSubmit={this.handleSubmit}>
                    <Field>
                      <Control>
                        <Input
                          name="email"
                          isSize={size}
                          onChange={this.handleChange}
                          placeholder="Email"
                          value={email}
                        />
                      </Control>
                    </Field>
                    <Field>
                      <Control isExpanded>
                        <Input
                          name="password"
                          isSize={size}
                          onChange={this.handleChange}
                          type="password"
                          placeholder="password"
                          value={password}
                        />
                      </Control>
                    </Field>
                    <Field>
                      <Control>
                        <Button
                          type="submit"
                          isSize={size}
                          isColor="info"
                          isBlock
                          isFullWidth
                        >
                          { buttonText }
                        </Button>
                      </Control>
                    </Field>
                  </form>
                </Box>
              </Column>
            </Columns>
          </Container>
        </HeroBody>
      </Hero>
    );
  }
}

TaskForm.propTypes = {
  submitAction: PropTypes.func.isRequired,
  buttonText:   PropTypes.string,
  size:         PropTypes.string,
};

TaskForm.defaultProps = {
  size:       'large',
  buttonText: 'Log In',
};
