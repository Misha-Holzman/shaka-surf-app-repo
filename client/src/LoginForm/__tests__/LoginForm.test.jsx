/* eslint-env jest */
import React from 'react';
import renderer from 'react-test-renderer';
import LoginForm from '..';

describe('LoginForm', () => {

  it('should render', () => {
    const fn = jest.fn();
    const component = renderer.create(
      <LoginForm submitAction={fn} />
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
