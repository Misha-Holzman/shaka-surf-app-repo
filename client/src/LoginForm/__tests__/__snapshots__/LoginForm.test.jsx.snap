// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm should render 1`] = `
<section
  className="hero is-fullheight is-light"
  style={
    Object {
      "margin-top": "-60px",
    }
  }
>
  <div
    className="hero-body"
  >
    <div
      className="container has-text-centered"
    >
      <div
        className="columns"
      >
        <div
          className="column is-4 is-offset-4"
        >
          <div
            className="box"
          >
            <h1
              className="title"
            >
              Hiya!
            </h1>
            <h2
              className="subtitle"
            >
              Letâ€™s Get You logged in...
            </h2>
            <form
              onSubmit={[Function]}
            >
              <div
                className="field"
              >
                <div
                  className="control"
                >
                  <input
                    className="input is-large"
                    name="email"
                    onChange={[Function]}
                    placeholder="Email"
                    type="text"
                    value=""
                  />
                </div>
              </div>
              <div
                className="field"
              >
                <div
                  className="control is-expanded"
                >
                  <input
                    className="input is-large"
                    name="password"
                    onChange={[Function]}
                    placeholder="password"
                    type="password"
                    value=""
                  />
                </div>
              </div>
              <div
                className="field"
              >
                <div
                  className="control"
                >
                  <button
                    className="button is-info is-large is-fullwidth"
                    isBlock={true}
                    type="submit"
                  >
                    Log In
                  </button>
                </div>
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;
